src/backend/TxCache.h

```diff
@@ -0,0 +1,56 @@
+#ifndef CLIO_TXCACHE_H_INCLUDED
+#define CLIO_TXCACHE_H_INCLUDED
+
+#include <ripple/basics/base_uint.h>
+#include <ripple/basics/hardened_hash.h>
+#include <backend/Types.h>
+#include <map>
+#include <mutex>
+#include <shared_mutex>
+#include <utility>
+#include <vector>
+
+#define NUM_LEDGERS_CACHED 10
+
+namespace Backend {
+class TxCache
+{
+    // basically a ring buffer
+    // array of maps of transaction hashes -> TransactionAndMetadata objects
+    // replace the oldest ledger in cache when inserting a new ledger
+    std::array<
+        std::map<ripple::uint256, TransactionAndMetadata>,
+        NUM_LEDGERS_CACHED>
+        cache_;
+    mutable std::shared_mutex mtx_;
+    uint32_t latestSeq_ = 0;
+    std::atomic_int tail_;
```

Comment:
I think this needs to be initialized

Reply:
Can you clarify what needs to be initialized? `tail_`?